### PR TITLE
Upgrade torch to 1.13.1 and check the cluster version before uploading a NLP model. 

### LINKED ISSUE #481: Detect incompatible PyTorch versions by the Elasticsearch
#479 shows that new versions of PyTorch will sometimes be incompatible with older versions. In this case a Torchscript model traced in version 1.12 of PyTorch cannot be evaluated in version 1.11. PyTorch will continue to be upgraded in both the Elastic stack and Eland in order to pick up new features, performance improvements and security fixes, inevitably this means that at some point certain versions of Eland will not work with Elasticsearch.

Currently, a user who has not read the release notes will only discover the incompatibility when they evaluate the model and get an error similar to 

```
Attempted to read a PyTorch file with version 10, but the maximum supported version for reading is 6. Your PyTorch installation may be too old.
```

We can do a better job of detecting the issue early in the `eland_import_hub_model` script. 
- If the Eland version is >= Elasticsearch version then Eland knows if the PyTorch versions are compatible or not.
- Else if the Eland version is < Elasticsearch version then Eland cannot know if the model it will create is future compatible with Elasticsearch. In which case politely decline to do the work and ask the user to upgrade Eland

This should be done when PyTorch is upgraded to 1.12 or greater.

### PR DESCRIPTION
**Pull request by davidkyle**
In the 8.7 release of Elasticsearch PyTorch is upgraded to version 1.13.1 (elastic/ml-cpp#2430). The change means the same version of PyTorch is used when tracing the model in Eland as evaluating it in Elasticsearch.

Some models traced in version 1.13 of PyTorch cannot be evaluated in version 1.9 or earlier. Because of this Eland will now not upload a model to a pre 8.7 Elasticsearch.

`transformers` compatibility is updated to include the latest release. PyTorch 1.13 is compatible with Python 3.10, the docs are updated to reflect this.

Closes #481

### CODE REVIEW

**Comment by davidkyle** [comment]
CI is failing because version 8.7 of the elasticsearch python client is not released yet

**Comment by davidkyle** [comment]
> CI is failing because version 8.7 of the elasticsearch python client is not released yet

I've reverted back to version 8.3. Version 8.7 is recommended but not necessary

**Comment by JonathanHourany** [comment]
Hello! Just wanted to check in on this branch. I was toying around with eland earlier and found myself in a little bit of a dependency struggle so I'm glad to see the bumps to `torch` and Python 3.10. Any thoughts on when this might get merged to master and release?

**Comment by davidkyle** [comment]
Hi @JonathanHourany this change will be in the next release: version 8.8. Please watch the PR as I will try to get it merged soon.

If you are using Eland to install models in Elasticsearch but having problems with dependencies I recommend using the Docker image. I tend to use the Docker image to simplify my setup. It is easy to get started, see the instructions in the README: https://github.com/elastic/eland#docker

File: requirements-dev.txt [hunk 1]
```diff
@@ -14,13 +14,13 @@ scikit-learn>=0.22.1,<2
 xgboost>=0.90,<2
 lightgbm>=2,<4
 
-# PyTorch doesn't support Python 3.10 yet (pytorch/pytorch#66424)
+# PyTorch doesn't support Python 3.11 yet (pytorch/pytorch#86566)
 
 # Elasticsearch uses v1.11.0 of PyTorch
```
Comment (sethmlarson) [inline requirements-dev.txt:19]: Update this comment (maybe with a reference to the ml-cpp PR?)

File: tests/ml/pytorch/test_pytorch_model_upload_pytest.py [hunk 2]
```diff
@@ -38,7 +38,7 @@
 
 pytestmark = [
     pytest.mark.skipif(
-        ES_VERSION < (8, 0, 0),
+        ES_VERSION < (8, 7, 0),
         reason="This test requires at least Elasticsearch version 8.0.0",
```
Comment (sethmlarson) [inline tests/ml/pytorch/test_pytorch_model_upload_pytest.py:42]: Update `reason` field

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Looks good, I only had two small comments:

**Review by sethmlarson — APPROVED** [review]

**Review by joshdevins — APPROVED** [review]
LGTM. I only did a visual pass, I didn't run anything.

**Comment by JonathanHourany** [comment]
> Hi @JonathanHourany this change will be in the next release: version 8.8. Please watch the PR as I will try to get it merged soon.

That's great to hear @davidkyle , thanks for the update!

**Pull request merged**